src/lib.rs

```diff
@@ -201,8 +201,16 @@ impl<S> WebSocketStream<S> {
     }
 }
 
+impl<T> Drop for WebSocketStream<T> where T: AsyncRead + AsyncWrite
+{
+    fn drop( &mut self )
+    {
+        let _ = self.inner.close(None);
```

Comment:
This won't work properly. It only queues messages for sending and does not guarantee that they will be sent. The network connection will be closed prior to performing close handshake.